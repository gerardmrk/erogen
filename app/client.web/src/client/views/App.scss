.main {
  min-height: 100%;
  display: grid;
  grid-template-rows: 100%;
  grid-template-columns: 100%;
  background-color: crimson;
}

.container {
  min-height: 100%;
  background-color: #f9f9f9;

  & > div {
    // margin: 2em auto;
    height: 100%;
  }
}
