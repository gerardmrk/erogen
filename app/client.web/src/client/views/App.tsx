import * as React from "react";
import { hot } from "react-hot-loader";
import { withRouter, RouteComponentProps } from "react-router";

type Props = {};

type State = {};

export class App extends React.Component<
  Props & RouteComponentProps<{}>,
  State
> {
  public render() {
    return <div>{"App"}</div>;
  }
}

export default hot(module)(withRouter(App));
