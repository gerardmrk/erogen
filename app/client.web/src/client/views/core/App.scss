@import "~@client/views/styles.colors.scss";
@import "~@client/views/styles.dimens.scss";

body {
  color: $text-color;
  line-height: 20px;
  font-size: 14px;
  font-weight: 400;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen,
    Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 500;
  }

  p {
    margin-top: 1px;
    margin-bottom: 1px;
    letter-spacing: -0.05px;
  }

  a {
    color: inherit;
    text-decoration: none;
  }
}

.main {
  min-height: 100%;
  display: grid;
  grid-template-rows: $app-header-height auto $app-footer-height;
  grid-template-columns: 100%;
}
