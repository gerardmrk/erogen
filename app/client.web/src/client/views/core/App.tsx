import "@client/views/theme/semantic.less";
import * as React from "react";
import { hot } from "react-hot-loader";
import loadable from "@loadable/component";
import styles from "./App.scss";
import GlobalLoader from "./GlobalLoader";

const AppHeader = loadable(() => import("@client/views/core/AppHeader"));
const AppContent = loadable(() => import("@client/views/core/AppContent"));
const AppFooter = loadable(() => import("@client/views/core/AppFooter"));

type Props = {};

type State = {};

export class App extends React.Component<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <GlobalLoader />
        <AppHeader />
        <AppContent />
        <AppFooter />
      </div>
    );
  }
}

export default hot(module)(App);
