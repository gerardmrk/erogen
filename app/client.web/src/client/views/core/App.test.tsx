import * as React from "react";
import { create, ReactTestRenderer, ReactTestInstance } from "react-test-renderer"; // prettier-ignore

import { App } from "./App";

const Component = App;

describe("<App/>", () => {
  let renderer: ReactTestRenderer;
  let component: ReactTestInstance;

  beforeEach(() => {
    renderer = create(<Component />);
    component = renderer.getInstance() as ReactTestInstance;
  });

  it("renders ok", () => {
    expect(component).toBeDefined();
  });
});
