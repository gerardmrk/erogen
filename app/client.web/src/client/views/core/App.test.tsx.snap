// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> snapshot 1`] = `
<div
  className="main"
>
  <div
    className="main"
  />
  <header
    className="main"
  >
    <div
      className="ui container container"
    >
      <div
        className="logo"
      >
        <div
          className="main"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="brand logo"
              src="logo.svg"
            />
          </a>
        </div>
      </div>
      <div
        className="main"
      >
        <nav
          className="main"
        >
          <div
            className="otherNavsGroup"
          >
            <div
              className="navBox"
            >
              <a
                aria-current={null}
                href="/about/product"
                onClick={[Function]}
              >
                <span>
                  navs.public.product
                </span>
              </a>
            </div>
            <div
              className="navBox"
            >
              <a
                aria-current={null}
                href="/blog"
                onClick={[Function]}
              >
                <span>
                  navs.public.blog
                </span>
              </a>
            </div>
            <div
              className="navBox"
            >
              <a
                aria-current={null}
                href="/documentation"
                onClick={[Function]}
              >
                <span>
                  navs.public.docs
                </span>
              </a>
            </div>
            <div
              className="navBox"
            >
              <a
                aria-current={null}
                href="/support"
                onClick={[Function]}
              >
                <span>
                  navs.public.support
                </span>
              </a>
            </div>
          </div>
          <div
            className="defaultNavsGroup"
          >
            <div
              className="navBox"
            >
              <a
                aria-current={null}
                href="/login"
                onClick={[Function]}
              >
                <span>
                  navs.public.login
                </span>
              </a>
            </div>
            <div
              className="navBox"
            >
              /
            </div>
            <div
              className="navBox"
            >
              <a
                aria-current={null}
                href="/register"
                onClick={[Function]}
              >
                <span>
                  navs.public.register
                </span>
              </a>
            </div>
          </div>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="main"
  >
    <div
      className="routeContainer"
    >
      <div
        className="main"
      >
        <div
          className="main"
        >
          <section
            className="main"
          >
            <div
              className="ui container container"
            >
              <div
                className="canvas"
              >
                <div
                  className="leftPane"
                >
                  <h1>
                    <em>
                      hero.heading
                    </em>
                  </h1>
                  <div
                    className="fluff"
                  >
                    <br />
                    <p>
                      Alpha bravo charlie delta echo foxtrot golf
                    </p>
                    <br />
                    <p>
                      Hotel india juliet kilo lima mike november oscar papa
                    </p>
                    <br />
                    <p>
                      Quebec romeo sierra tango unicorn victor whiskey x-ray yankee zulu
                    </p>
                  </div>
                </div>
                <div
                  className="rightPane"
                >
                  <img
                    src="astro.svg"
                  />
                </div>
              </div>
            </div>
          </section>
          <section
            className="main"
          >
            <div
              className="ui container container"
            >
              <div
                className="brandCarousel"
              >
                <div
                  className="brandItem"
                >
                  <p>
                    Smashing
                  </p>
                </div>
                <div
                  className="brandItem"
                >
                  <p>
                    Pumpkins
                  </p>
                </div>
                <div
                  className="brandItem"
                >
                  <p>
                    Radiohead
                  </p>
                </div>
                <div
                  className="brandItem"
                >
                  <p>
                    Chilli
                  </p>
                </div>
                <div
                  className="brandItem"
                >
                  <p>
                    Peppers
                  </p>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="main"
  >
    <div
      className="ui container container"
    >
      <div
        className="main"
      >
        <div
          className="container"
        >
          
        </div>
      </div>
      <div
        className="main"
      >
        <div
          className="container"
        />
      </div>
      <div
        className="footerEnd"
      >
        <div
          className="main"
        >
          <div
            className="container"
          >
            copyright
          </div>
        </div>
        <div
          className="main"
        >
          <div
            aria-expanded={false}
            className="ui fluid selection dropdown"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            role="listbox"
            tabIndex={0}
          >
            <div
              aria-atomic={true}
              aria-live="polite"
              className="text"
              role="alert"
            />
            <i
              aria-hidden="true"
              className="dropdown icon"
              onClick={[Function]}
            />
            <div
              className="menu transition"
            >
              <div
                aria-checked={false}
                aria-selected={true}
                className="selected item"
                onClick={[Function]}
                role="option"
                style={
                  Object {
                    "pointerEvents": "all",
                  }
                }
              >
                <span
                  className="text"
                >
                  languages.en
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
